scylla-cql/src/frame/response/result.rs

```diff
@@ -619,14 +619,30 @@ pub enum Result {
     SchemaChange(SchemaChange),
 }
 
-fn deser_table_spec(buf: &mut &[u8]) -> StdResult<TableSpec<'static>, TableSpecParseError> {
-    let ks_name = types::read_string(buf)
-        .map_err(TableSpecParseError::MalformedKeyspaceName)?
-        .to_owned();
-    let table_name = types::read_string(buf)
-        .map_err(TableSpecParseError::MalformedTableName)?
-        .to_owned();
-    Ok(TableSpec::owned(ks_name, table_name))
+// Only allocates a new `TableSpec` if one is not yet given.
+fn deser_table_spec<'frame: 'spec, 'spec>(
```

Comment:
Can we close https://github.com/scylladb/scylla-rust-driver/pull/956 then?

Reply:
If we are going to store a borrowed `TableSpec` in each `ColumnSpec`, then I don't think we need to remove it from `ColumnSpec`.

Reply:
This is not the final state. In a follow-up of the deserialization refactor, I'm going to decouple `TableSpec` from `ColumnSpec`. The reason for doing this in a follow-up is to prove that such change won't break the public API, so an opposite change won't, too, which is what @Lorak-mmk believes is important in case Scylla/Cassandra once introduce joins to CQL.

Reply:
Ok,, makes sense